processing/src/main/java/org/apache/druid/segment/column/ColumnIndexCapabilities.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.column;
+
+import org.apache.druid.query.filter.ColumnIndexSelector;
+
+/**
+ * Sort of like {@link ColumnCapabilities}, except for indexes supplied by {@link ColumnIndexSelector}, provides
+ * information for how query processing may use indexes.
+ */
+public interface ColumnIndexCapabilities
```

Comment:
> ColumnIndexCapabilities belongs to a column and is acting like it represents all things from that column, it cannot, however do that.

This isn't true, `getIndexCapabilities` takes both a column, and a specific class implementation of the index, so these are the capabilities of a specific index of the column. I guess I wasn't considering the case where an index might sometimes produce exact matches and sometimes not, since in my mind these wouldn't be mixed within the same index, but I wasn't thinking clever enough about it and these cases probably do exist.